crates/router/src/core/webhooks/incoming.rs

```diff
@@ -220,31 +220,215 @@ async fn incoming_webhooks_core<W: types::OutgoingWebhookType>(
+        &body,
+        &request_details,
+        merchant_connector_account.as_ref(),
+    )
+    .await?;
+
+    let (event_type, source_verified_via_ucs, webhook_transform_data, final_request_body) =
+        match webhook_result {
+            WebhookProcessingResult::ProcessFurther {
+                event_type,
+                source_verified,
+                transform_data,
+                decoded_body,
+            } => (event_type, source_verified, transform_data, decoded_body),
+            WebhookProcessingResult::EarlyReturn {
+                response,
+                webhook_tracker,
+                serialized_request,
+            } => {
+                // Early return for unsupported event types
+                return Ok((*response, webhook_tracker, serialized_request));
+            }
+        };
+
+    // Update request_details with the appropriate body (decoded for non-UCS, raw for UCS)
+    let final_request_details = if let Some(ref decoded_body) = final_request_body {
+        IncomingWebhookRequestDetails {
+            method: request_details.method.clone(),
+            uri: request_details.uri.clone(),
+            headers: request_details.headers,
+            query_params: request_details.query_params.clone(),
+            body: decoded_body,
+        }
+    } else {
+        request_details // UCS path uses raw body
+    };
+
+    logger::info!(event_type=?event_type);
+
+    // Check if webhook should be processed further
+    let is_webhook_event_supported = !matches!(
+        event_type,
+        webhooks::IncomingWebhookEvent::EventNotSupported
+    );
+    let is_webhook_event_enabled = !utils::is_webhook_event_disabled(
+        &*state.clone().store,
+        connector_name.as_str(),
+        merchant_context.get_merchant_account().get_id(),
+        &event_type,
+    )
+    .await;
+
+    let process_webhook_further = is_webhook_event_enabled && is_webhook_event_supported;
+    logger::info!(process_webhook=?process_webhook_further);
+
+    let flow_type: api::WebhookFlow = event_type.into();
+    let mut event_object: Box<dyn masking::ErasedMaskSerialize> = Box::new(serde_json::Value::Null);
+
+    let webhook_effect =
+        if process_webhook_further && !matches!(flow_type, api::WebhookFlow::ReturnResponse) {
+            // Process main webhook business logic
+            match process_webhook_business_logic(
+                &state,
+                req_state,
+                &merchant_context,
+                &connector,
+                &connector_name,
+                event_type,
+                source_verified_via_ucs,
+                &webhook_transform_data,
+                &final_request_details,
+                is_relay_webhook,
+            )
+            .await
+            {
+                Ok(response) => {
+                    // Extract event object for serialization
+                    event_object = if let Some(transform_data) = webhook_transform_data {
```

Comment:
Avoid deep nested if else structure, have handler function to handle response handling, and for option use match and avoid using if let some syntax, chaining and writing in idiomatic way follow those.